tools/CodeGenerator/KnownHeaders.cs

```diff
@@ -408,6 +433,7 @@ protected override void SetValueFast(string key, StringValues value)
         protected override bool AddValueFast(string key, StringValues value)
         {{{(loop.ClassName == "HttpResponseHeaders" ? @"
             ValidateHeaderCharacters(value);" : "")}
+            var isNotNull = value.Count > 0;
```

Comment:
Is this the fastest way to check for emptiness?

Reply:
I think so 😢 
```csharp
int Count => _value != null ? 1 : (_values?.Length ?? 0);
```
There is a pass by value static method; but looks like that does more?
```csharp
public static bool IsNullOrEmpty(StringValues value)
{
    if (value._values == null)
    {
        return string.IsNullOrEmpty(value._value);
    }
    switch (value._values.Length)
    {
        case 0: return true;
        case 1: return string.IsNullOrEmpty(value._values[0]);
        default: return false;
    }
}
```

Reply:
We should optimize this. Looks very branchy 😄

Reply:
```csharp
public bool IsNull()
{
    var values = _values;
    return (_value != null) ? 
        false : 
        ((values == null || values.Length == 0) ? 
            true : 
            false);
}
```
? 😟

Reply:
haz idea

Reply:
https://github.com/aspnet/Common/pull/323